### ISSUE TITLE
faulty conversion of x-ray to Jaeger

### ISSUE THREAD

**Issue by HashemTaheriSonos** [open-telemetry/opentelemetry-collector#3501]
**Describe the bug**
I'm exporting traces to both AWS and Jaeger. I set up an otel-collector which receives in the x-ray format and export to Jaeger. The traces are fully shown in the Jaeger but the dependency graph is not showing anything and cannot compute the tree. as I checked that's because these traces, do not have any `references` value set, that's why from the jaeger side [it cannot compute the tree.](https://github.com/jaegertracing/spark-dependencies/blob/1e1507ae3b4f5de90d49e6af27a24288ffab6bd3/jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/SpansToDependencyLinks.java#L77)

so I think the way the traces are getting converted from x-ray format to jaeger might have a bug.

**Steps to reproduce**
use ADOT javaagent with a java app to be able to produce traces in x-ray format
then send these traces to the ADOT collector
from ADOT collector redirect it to an OTEL collector
and from OTEL collector export it to the Jaeger

**What did you expect to see?**
Having a dependency graph (the `references` field of the traces shouldn't be null)

**What did you see instead?**
No dependency graph - the `references` value of spans are zero (null)

**What version did you use?**
OTEL collector 0.28.0
ADOT collector 0.10.0
javaagent - version: 1.2.0-aws
Jaeger 1.22.0

**Comment by rakyll** [comment]
cc @anuraaga

**Comment by anuraaga** [comment]
Hi @HashemTaheriSonos - are you using the OTLP receiver on the collector? If you're using jaeger receiver, can you try OTLP?

X-ray is only a factor when exporting to from the collector, there is no input "x-ray format" to convert from to jaeger. So the issue is probably either in jaeger exporter or the instrumentation

**Comment by HashemTaheriSonos** [comment]
@anuraaga Thanks for your response, I'm not using Jaeger receiver, just in the last step I'm exporting as Jaeger.
I see... true, it might be in the Jaeger exporting part which cause this issue.
The flow is like below:

```
Client
         |
Export traces thanks to java-agent (aws)
         |
ADOT agent:
    receivers: [otlp,awsxray]
    processors: [batch/traces]
    exporters: [awsxray, otlp]
        |
        |
ADOT collector
    receivers: [otlp,awsxray]
    processors: [batch/traces]
    exporters: [awsxray, otlp]
       |
       |
OTEL collector
      receivers: [otlp]
      processors: [batch]
      exporters: [jaeger]
      |
      |
Jaeger Collector
```

**Comment by anuraaga** [comment]
The awsxray receiver is only meant to be used with x-ray SDK, so if you are using Otel javaagent you don't need it in your config. I'm not sure if between adot agent and collector in your diagram whether it's being used or not but just in case can you try disabling it? I guess it might not change anything though so would be good to get help from someone familiar with jaeger.

**Comment by HashemTaheriSonos** [comment]
@anuraaga True, but i'm using the [java-agent by aws](https://github.com/aws-observability/aws-otel-java-instrumentation), not the otel javaagent, so I'm quite sure it's in use, (if I disable it basically i'm not going to have traces in aws side, for instance in the X-ray dashboard)

but yup, between ADOT Agent and Collector, I can try to remove the awsxray and keep just the OTLP, to see if it will affect the issue. as you said I don't think it will solve the issue but worth trying.

**Comment by anuraaga** [comment]
@HashemTaheriSonos Sorry for the confusion - the AWS Otel java agent exports with OTLP, it doesn't talk to the xrayreceiver. You can and should disable it.

The only real difference between the AWS Otel java agent and normal otel one is it generates IDs that are compatible with x-ray (and any other backend since they are a subset).

**Comment by HashemTaheriSonos** [comment]
@anuraaga no worries at all... Okay, I see... I didn't know that it exports with OTLP, so makes sense, I will disable it and try it to see the result.
